pkg/kratos/service.go

```diff
@@ -387,6 +393,78 @@ func (s *Service) UpdateRecoveryFlow(
 	return returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) executeIdentifierFirstUpdateLoginRequest(
+	ctx context.Context, flow string, csrfToken, identifier string, cookies []*http.Cookie,
+) (*http.Response, error) {
+	form := url.Values{}
+	form.Set("csrf_token", csrfToken)
+	form.Set("identifier", identifier)
+	form.Set("method", "identifier_first")
+
+	if s.loginURL == nil {
+		return nil, fmt.Errorf("missing kratos login url")
+	}
+	u := *s.loginURL
+	q := u.Query()
+	q.Set("flow", flow)
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), strings.NewReader(form.Encode()))
+	if err != nil {
+		s.logger.Errorf("failed to create http request: %s", err)
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	for _, c := range cookies {
+		req.AddCookie(c)
+	}
+
+	client := *s.httpClient
+	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		// Kratos returns 303 for identifier first flows. We disable these automatic redirects
+		// in order to let the handler process the BrowserLocationChangeRequired and change to 200
+		return http.ErrUseLastResponse
+	}
+	return client.Do(req)
+}
+
+func (s *Service) UpdateIdentifierFirstLoginFlow(
+	ctx context.Context, flow string, body kClient.UpdateLoginFlowBody, cookies []*http.Cookie,
+) (*BrowserLocationChangeRequired, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.UpdateIdentifierFirstLoginFlow")
+	defer span.End()
+
+	csrfToken, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetCsrfTokenOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing csrf token")
+	}
+
+	identifier, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetIdentifierOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing identifier")
+	}
+
+	resp, err := s.executeIdentifierFirstUpdateLoginRequest(ctx, flow, *csrfToken, *identifier, cookies)
+	if err != nil {
+		s.logger.Errorf("kratos request failed: %s", err)
+		return nil, nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusSeeOther {
+		location := resp.Header.Get("Location")
+		return &BrowserLocationChangeRequired{RedirectTo: &location}, resp.Cookies(), nil
+	}
+
+	if resp.StatusCode == http.StatusBadRequest {
+		err = s.getUiError(resp.Body)
+		return nil, nil, err
+	}
+
+	return nil, nil, fmt.Errorf("unexpected status: %d", resp.StatusCode)
```

Comment:
suggestion: log something here, not sure when this can happen but we should have some info about it